### ISSUE TITLE
Toml parsing fails if it uses workspace keys

### ISSUE THREAD

**Issue by SUPERCILEX** [cargo-public-api/cargo-public-api#175]
This:

```toml
[package]
version.workspace = true
```

Causes the following crash
```
thread 'api' panicked at 'called `Result::unwrap()` on an `Err` value: CargoTomlError(Parse(Error { inner: ErrorInner { kind: Custom, line: Some(2), col: 20, at: Some(50), message: "invalid type: map, expected a string", key: ["package", "version"] } }))', fuc_engine/tests/api.rs:10:82
```

Not using `*.workspace = true` fixes the issue.

**Comment by Emilgardis** [comment]
This is a upstream issue on `cargo_toml`, need to wait for `cargo_toml@0.13.0`, there is currently an [alpha](https://crates.io/crates/cargo_toml/0.13.0-alpha.1) out that has started the implementation of this

**Comment by Enselic** [comment]
I think I prefer depending on alphas over crashing, so I created a PR with this bump: #179

Any objections to us depending on this alpha for a while?

**Comment by SUPERCILEX** [comment]
No objections. I also prefer having something working right now knowing it'll break in the future than nothing. :)

**Comment by Emilgardis** [comment]
Only important thing is that if we do a release with the alpha, that we also mark that release as a rc/alpha

**Comment by Enselic** [comment]
Good point. There is no compatibility guarantee of pre-releases, so `cargo install` might suddenly fail if we depend on an alpha version. [This](https://doc.rust-lang.org/cargo/reference/resolver.html#pre-releases) says 

> it is recommended to not use pre-release dependencies in a library if your library is not also a pre-release.

so I think we should be more careful than I initially thought. And I don't want to manage pre-releases of this project itself.

I'll close the PR for now, and then we can re-open it when there is a stable release of `cargo_toml`. At least we now know the fix works.

**Comment by Enselic** [comment]
0.13.0 has been released :partying_face: 

PR with fix and regression test: #186 

(Same as old PR, but I was unable to re-open it for some reason)

**Comment by Enselic** [comment]
Closed by https://github.com/Enselic/cargo-public-api/pull/186
